File: app/src/main/java/com/punam/weeksixassignmentone/StudentData.java
package com.punam.weeksixassignmentone;

import com.punam.weeksixassignmentone.model.Student;

import java.util.ArrayList;

public class StudentData {
    ArrayList<Student> list = new ArrayList<>();
    private static StudentData data = new StudentData();

    private StudentData(){}

    public  static StudentData get(){
        return data;
    };

    public  ArrayList<Student> List (){
        return list;
    }
}

File: app/src/main/java/com/punam/weeksixassignmentone/StudentData.kt
package com.punam.weeksixassignmentone

import com.punam.weeksixassignmentone.model.Student
import java.util.*

class StudentData private constructor() {
    var list = ArrayList<Student>()
    fun List(): ArrayList<Student> {
        return list
    }

    companion object {
        private val data = StudentData()
        fun get(): StudentData {
            return data
        }
    }
}